build.gradle

```diff
@@ -55,13 +55,14 @@ ext {
 		osgiVersion = "${version}.RELEASE"
 		println "$version is a release, will use $osgiVersion for bnd"
 	}
-	else if (versionNumber.qualifier.equalsIgnoreCase("SNAPSHOT")) {
+	else if (!versionNumber.qualifier.equalsIgnoreCase("SNAPSHOT")) {
+		osgiVersion = "${versionNumber.major}.${versionNumber.minor}.${versionNumber.micro}.${versionNumber.qualifier}"
+		println "$version is neither release nor snapshot, will use $osgiVersion for bnd"
 		osgiVersion = "${versionNumber.major}.${versionNumber.minor}.${versionNumber.micro}.BUILD-SNAPSHOT"
-		println "$version is a snapshot, will use $osgiVersion for bnd"
 	}
 	else {
-		osgiVersion = "${versionNumber.major}.${versionNumber.minor}.${versionNumber.micro}.${versionNumber.qualifier}"
-		println "$version is neither release nor snapshot, will use $osgiVersion for bnd"
+		osgiVersion = version
```

Comment:
@simonbasle  I think the issue with the -snapshot version was here. BND is processing -snaptshot for the Bundle-Version attribute AFTER it's processing the exports.

Therefore at the point of export OSGi sees following version:
3.4.1-SNAPSHOT

which OSGi can't parse as a valid OSGi version because OSGi would require a qualifier with a dot (.):
3.4.1.SNAPSHOT

When the Bundle-Version is no valid OSGi version the qualifier is not removed for package exports as BND does for valid OSGi versions. So for Export-Package BND would convert the Bundle-Version "3.4.1.SNAPSHOT" into "3.4.1" for package exports (valid OSGi version, stripping qualifier) but wen it's not a valid OSGi version as with "3.4.1-SNAPSHOT" the version is kept as it is so it results in "3.4.1-SNAPTSHOT" package versions.

I would say that this is really inconvenient and I will create a BND issue for that. 

As a result of that to get proper Export-Package versions we would need to define a custom osgiVersion anyways because we would need to convert the maven version 3.4.1-SNAPSHOT into 3.4.1.SNAPSHOT here therefore using the bnd instruction -snapshot is not really worth it because we have to do the version conversion anyway.